flojoy/flojoy_cloud.py

```diff
@@ -0,0 +1,409 @@
+                            "r": red_channel,
+                            "g": green_channel,
+                            "b": blue_channel,
+                            "a": alpha_channel,
+                        }
+                    },
+                    cls=NumpyEncoder,
+                )
+
+        return payload
+
+    def store_dc(self, data, dc_type):
+        """
+        A method that stores a formatted data payload onto the Flojoy cloud.
+        """
+        url = "https://cloud.flojoy.ai/api/v1/dcs/"
+        payload = self.create_payload(data, dc_type)
+        response = requests.request("POST", url, headers=self.headers, data=payload)
+        return json.loads(response.text)
+
+    def fetch_dc(self, dc_id):
+        """
+        A method that retrieves DataContainers from the Flojoy cloud.
+        """
+        url = f"https://cloud.flojoy.ai/api/v1/dcs/{dc_id}"
+        response = requests.request("GET", url, headers=self.headers)
+        response = json.loads(response.text)
+        check_deserialize(response)
+        return response
+
+    def to_python(self, dc):
+        """
+        A method that converts data from DataContainers into pythonic
+        data types like Pillow for images.
+        """
+        dc_type = dc["dataContainer"]["type"]
+        match dc_type:
+            case "OrderedPair":
+                df = pd.DataFrame(dc["dataContainer"])
+                return df.drop(columns=["type"])
+            case "OrderedTriple":
+                df = pd.DataFrame(dc["dataContainer"])
+                return df.drop(columns=["type"])
+            case "DataFrame" | "Matrix" | "Grayscale":
+                df = pd.DataFrame(dc["dataContainer"]["m"])
+                return df
+            case "Scalar":
+                return float(dc["dataContainer"]["c"])
+            case "Image":
+                image = dc["dataContainer"]
+                r = image["r"]
+                g = image["g"]
+                b = image["b"]
+                if "a" in image:
+                    a = image["a"]
+                    img_combined = np.stack((r, g, b, a), axis=2)
+                    return Image.fromarray(np.uint8(img_combined)).convert("RGBA")
+                else:
+                    img_combined = np.stack((r, g, b), axis=2)
+                    return Image.fromarray(np.uint8(img_combined)).convert("RGB")
+
+    def list_dcs(self, size=10):
+        """
+        A method that lists the number of DataContainers specified.
+
+        If the number of DataContainers is less than the specified number,
+        an error will be thrown.
+        """
+        url = f"https://cloud.flojoy.ai/api/v1/dcs/?inbox=true&size={size}"
+        response = requests.request("GET", url, headers=self.headers, data="")
+        response = json.loads(response.text)
+        try:
+            response = response["data"]
+        except KeyError:
+            print("Size larger than the number of DataContainers.")
```

Comment:
Does this endpoint actually return a response with no data field sometimes?

Reply:
If you try to list more DC's than there are, it returns a json with an error.

Reply:
Hmm, I don't seem to have this issue (using postman), I set size to 10000 even though I have 2 DCs in the inbox and it worked fine